code/obj/item/clothing/hats.dm

```diff
@@ -1420,6 +1428,41 @@ TYPEINFO(/obj/item/clothing/head/that/gold)
 
 		..()
 
+	proc/attempt_shock(mob/wearer, mob/target)
+		if(wearer != target && src.uses > 0 && target.a_intent == INTENT_HELP && target.zone_sel?.selecting == "head")
+			wearer.visible_message(SPAN_ALERT("[target] tries to pat [wearer] on the head, but gets shocked by [src]!"))
+			zap_headpatter(target)
+
+			src.uses = max(0, src.uses - 1)
+			if (src.uses < 1)
+				src.icon_state = splittext(src.icon_state,"-")[1]
+				src.item_state = splittext(src.item_state,"-")[1]
+				wearer.update_clothing()
+
+			if (src.uses <= 0)
+				wearer.show_text("The sunhat is no longer electrically charged.", "red")
+			else
+				wearer.show_text("The stunhat has [src.uses] charges left!", "red")
```

Comment:
should probably return `TRUE` to prevent the attackhand from going through anyways